File: java_solutions/404.sum-of-left-leaves.java
/*
 * @lc app=leetcode id=404 lang=java
 *
 * [404] Sum of Left Leaves
 */

class Solution {
    public int sumOfLeftLeaves(TreeNode root) {
        return root == null ? 0 : (root.left != null && root.left.left == null && root.left.right == null ? root.left.val : 0) + sumOfLeftLeaves(root.left) + sumOfLeftLeaves(root.right);
    }
}

File: kotlin_solutions/404.sum-of-left-leaves.kt
/*
 * @lc app=leetcode id=404 lang=kotlin
 *
 * [404] Sum of Left Leaves
 */

class Solution {fun sumOfLeftLeaves(root: TreeNode?): Int = root?.let { (it.left?.takeIf { l -> l.left == null && l.right == null }?.`val` ?: 0) + sumOfLeftLeaves(it.left) + sumOfLeftLeaves(it.right) } ?: 0}
